hnn_core/opt_toy_example/general.py

```diff
@@ -0,0 +1,365 @@
+"""Parameter optimization."""
+
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+import numpy as np
+
+from metrics import _rmse_evoked  # change path***
+
+from skopt import gp_minimize
+from scipy.optimize import fmin_cobyla
+
+
+class Optimizer:
```

Comment:
I'm in two minds whether we should have a private parent Optimizer class and two separate classes for cobyla and bayesian ...

Something like:

```py
class _Optimizer:

     def run(self):
         pass

     def assemble_constraints(self):
          pass

      def plot_convergence(self):
          # insert code for plotting convergence

     def set_params(self):
           pass

class BayesianOptimizer(_Optimizer):

       def run(self):
             # now call the function to run bayesian optimization

class CobylaOptimizer(_Optimizer):

       def run(self):
             # now call the function to run cobyla optimization
```

Reply:
Really good alternative, let's plan to discuss but for right now we should really focus on getting the basic functionality working and just merge this PR